hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/state/WritableKVStateBase.java

```diff
@@ -170,6 +174,17 @@ public final Set<K> modifiedKeys() {
         return modifications.keySet();
     }
 
+    /** {@inheritDoc}
+     * To add the comparator to the underlying data source, we create a new TreeMap with the Comparator
+     * This should be called on each writable store constructor until all other types of Key Comparator are implemented.
+     * */
+    @Override
+    public final void updateComparator(@NonNull final Comparator<K> comparator) {
+        final var tmp = new TreeMap<K, V>(comparator);
+        tmp.putAll(modifications);
+        this.modifications = tmp;
```

Comment:
I am afraid I am missing something very essential here. If we update the `Comparator`, we are not changing the order of the elements in the data structure but the order of modifications, right? Why is this necessary? I assume that if any code is executed deterministically and single-threaded, the modifications should always be in the same order.

Reply:
You have a linkedhashmap with put, remove options. If not sorted, how can you guarantee a deterministic order? We care about the writableKVState to be deterministic.

Reply:
A TreeMap is created in WritableKVStateBase constructor to enforce the Key order of modification when every service calls its' WritableXXXStore.put().